src/components/screens/group-dialog-screen.tsx

```diff
@@ -0,0 +1,394 @@
+
+    const filtered = allGroups.filter(group =>
+      group.name.toLowerCase().includes(value.toLowerCase())
+    );
+    setFilteredGroups(filtered);
+  };
+
+  /**
+   * Handle checkbox changes
+   *
+   * @param name group name
+   */
+  const handleCheckboxChange = (name: string) => {
+    const newCheckedItems = joinGroupCheckedItems.map(item => {
+      if (item.name.toLowerCase() === name.toLowerCase()) {
+        item.checked = !item.checked;
+        return item;
+      }
+      return item;
+    });
+
+    setJoinGroupCheckedItems(newCheckedItems);
+  };
+
+  /**
+   * Gets the check box value based on checkbox name
+   *
+   * @param name checkbox group name
+   */
+  const getCheckedValue = (name: string) => {
+    const checkbox = joinGroupCheckedItems.find(group => group.name.toLowerCase() === name.toLowerCase());
+
+    if (!checkbox) return;
+
+    return checkbox.checked;
+  };
+
+  /**
+   * Creates a new user group
+   */
+  const createGroup = async () => {
+    if (!newGroupName || !keycloak?.token) return;
+
+    const nameAlreadyExists = allGroups.some(groups => groups.name.toLowerCase() === newGroupName.toLowerCase());
+    if (nameAlreadyExists) {
+      setGroupNameError(true);
+      return;
+    }
+
+    setLoading(true);
+    try {
+      const createdGroup = await Api.getUserGroupsApi(keycloak.token).createUserGroup({
+        userGroup: {
+          name: newGroupName
+        }
+      });
+      dispatch(setUserGroups([...usersGroups, createdGroup]));
+      setDialogStatus(GroupDialogStatus.CREATE_DONE);
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.groupDialogsScreen.createGroup);
+    }
+
+    setNewGroupName("");
+    setGroupNameError(false);
+    setLoading(false);
+  };
+
+  /**
+   * Creates a new join group request
+   */
+  const joinGroup = async () => {
+    const groupsToJoin = joinGroupCheckedItems.filter(group => group.checked);
+    // TODO: currently set so user can only join a single group
+    const matchedGroup = allGroups.find(group => group.name.toLowerCase() === groupsToJoin[0].name.toLowerCase());
+
+    if (!matchedGroup || !keycloak?.token || !keycloak?.tokenParsed) return;
+
+    setJoinedGroup(matchedGroup?.name);
+
+    // TODO: currently set up to only join one group at a time
+    if (groupsToJoin.length === 1) {
+      if (!matchedGroup?.id) return;
+
+      setLoading(true);
+      try {
+        await Api.getGroupJoinRequestsApi(keycloak.token).createGroupJoinRequest({
+          groupId: matchedGroup.id,
+          groupJoinRequest: {
+            email: keycloak.tokenParsed.email,
+            groupId: matchedGroup.id
+          }
+        });
+
+        setDialogStatus(GroupDialogStatus.REQUEST_SENT);
+        setLoading(true);
+      } catch (error) {
+        errorContext.setError(strings.errorHandling.groupDialogsScreen.createGroupRequest);
+      }
+    } else {
+      errorContext.setError("Joining mutliple groups not yet supported");
+    }
```

Comment:
Reverse the condition:
```suggestion
    // TODO: currently set up to only join one group at a time
    if (groupsToJoin.length !== 1) {
      errorContext.setError("Joining multiple groups not yet supported");
      return;
    }

    if (!matchedGroup?.id) return;

    setLoading(true);

    try {
      await Api.getGroupJoinRequestsApi(keycloak.token).createGroupJoinRequest({
        groupId: matchedGroup.id,
        groupJoinRequest: {
          email: keycloak.tokenParsed.email,
          groupId: matchedGroup.id
        }
      });

      setDialogStatus(GroupDialogStatus.REQUEST_SENT);
      setLoading(true);
    } catch (error) {
       errorContext.setError(strings.errorHandling.groupDialogsScreen.createGroupRequest);
    }
```